src/eth_provider/provider.rs

```diff
@@ -427,10 +427,11 @@ where
         }
 
         let end_block = self.tag_into_block_number(newest_block).await?;
-        let end_block_plus = end_block + U64::from(1_u64);
+        let end_block = end_block.to::<u64>();
+        let end_block_plus = end_block + 1;
```

Comment:
nit: one of the reviewers on Reth suggested to use `saturating_add` and `saturating_sub`.